### PR TITLE
Remove trace/metrics exporter sinks.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: internal/collector/processor/exporter_processor.go [hunk 1]
```diff
@@ -19,18 +19,18 @@ import (
 
 	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
 	"github.com/census-instrumentation/opencensus-service/data"
-	"github.com/census-instrumentation/opencensus-service/exporter"
+	"github.com/census-instrumentation/opencensus-service/processor"
 )
 
 type exporterSpanProcessor struct {
-	tes exporter.TraceExporterSink
+	tes processor.TraceDataProcessor
```
Comment (songy23) [inline internal/collector/processor/exporter_processor.go]: s/tes/tdps/g

File: exporter/exporterwrapper/exporterwrapper.go [hunk 2]
```diff
@@ -29,41 +29,60 @@ import (
...
+// NewExporterWrapper returns a processor.TraceDataProcessor that converts OpenCensus Proto TraceData
+// to OpenCensus-Go SpanData and calls into the given trace.Exporter.
 //
 // This is a bootstrapping mechanism for us to re-use as many of
 // the OpenCensus-Go trace.SpanData exporters which were written
 // by various vendors and contributors. Eventually the goal is to
 // get those exporters converted to directly receive
 // OpenCensus Proto TraceData.
-func OcProtoSpansToOCSpanDataInstrumented(ctx context.Context, exporterName string, te trace.Exporter, td data.TraceData) (aerr error) {
+func NewExporterWrapper(exporterName string, ocExporter trace.Exporter) processor.TraceDataProcessor {
```
Comment (songy23) [inline exporter/exporterwrapper/exporterwrapper.go]: Why there's only Tracing exporter wrapper? Also consider renaming this to `NewTraceExporterWrapper`.

Comment (bogdandrutu) [reply exporter/exporterwrapper/exporterwrapper.go]: I had it as TraceExporterWrapper but decided to use ExporterWrapper for when we will wrap both.

File: internal/collector/processor/exporter_processor.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/collector/processor/exporter_processor.go]: Done.

File: exporter/exporterwrapper/exporterwrapper.go (hunk 2 above)
Comment (songy23) [reply exporter/exporterwrapper/exporterwrapper.go]: OK, make sense to me.

File: exporter/awsexporter/aws_xray.go [hunk 3]
```diff
@@ -146,13 +146,13 @@ func transformConfigToXRayOptions(axrCfg *awsXRayConfig) (xopts []xray.Option, e
...
 // It uniquely maintains
-func (axe *awsXRayExporter) ExportSpans(ctx context.Context, td data.TraceData) (xerr error) {
+func (axe *awsXRayExporter) ProcessTraceData(ctx context.Context, td data.TraceData) (xerr error) {
 	ctx, span := trace.StartSpan(ctx,
 		"opencensus.service.exporter.aws_xray.ExportSpans",
 		trace.WithSampler(trace.NeverSample()))
 
 	defer func() {
-		if xerr != nil {
+		if xerr != nil && span.IsRecordingEvents() {
```
Comment (pjanotti) [inline exporter/awsexporter/aws_xray.go]: q: I assume that this check doesn't affect the zpages, correct?

Comment (bogdandrutu) [reply exporter/awsexporter/aws_xray.go]: It doesn't, even for zpages the Span should record the events in order to be able to display them.

File: exporter/exporterwrapper/exporterwrapper.go [hunk 4]
```diff
@@ -29,41 +29,60 @@ import (
...
+	return &ocExporterWrapper{spanName: "opencensus.service.exporter." + exporterName + ".ExportTrace", ocExporter: ocExporter}
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply exporter/exporterwrapper/exporterwrapper.go]: We will do this later, trying to minimize the logical changes.

**Comment by bogdandrutu** [comment]
@pjanotti I need your approval :)

**Pull request merged**